R/lint.R

```diff
@@ -0,0 +1,143 @@
+# Retrieve the lifecycle status defined in each Rd file
+db_lifecycle <- function(db) {
+  lifecycle_patterns <- paste0("(?:",
+    paste(collapse = "|",
+      c("lifecycle::badge\\([\\\\]\"",
+        "rlang:::lifecycle\\([\\\\]\"",
+        "list\\(\"lifecycle-",
+        "https://www.tidyverse.org/lifecycle/#"
+      )),
+    ")([\\w-]+)"
+  )
+
+  desc <- lapply(db, asNamespace("tools")$.Rd_get_metadata, "description")
+  lapply(desc, function(x) {
+    res <- regexpr(lifecycle_patterns, x, perl = TRUE)
+    starts <- attr(res, "capture.start")
+    ends <- starts + attr(res, "capture.length") - 1
+    substring(x, starts, ends)
+  })
+}
+
+# Retrieve the functions listed in the usage for each Rd file in the database
+db_function <- function(db) {
+  usage <- lapply(db, asNamespace("tools")$.Rd_get_section, "usage")
+  lapply(usage, get_usage_function_names)
+}
+
+#' @param package The name of installed packages.
```

Comment:
installed packages singular?
```suggestion
#' @param package The name of an installed package.
```